### PR TITLE
Added collect signature for Consumer companion object (#1151)

### PR DESCRIPTION
**Pull request by paualarco**

### CODE REVIEW

File: monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala [hunk 1]
```diff
@@ -227,6 +227,17 @@ object Consumer {
...
+  /** A consumer that will return a collection of all the streamed
+   * values as a Scala `List`.
+   *
+   * Note that this method is functionally the same as the `.toListL` from [[Observable]]
+   *
+   * WARNING: For infinite streams the process will eventually
+   * blow up with an out of memory error.
+   */
+  def collect[A]: Consumer.Sync[A, List[A]] =
+    new FoldLeftConsumer[A, List[A]](() => List.empty[A], (l: List[A], elem: A) => l :+ elem)
```
Comment (Avasil) [inline monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala]: I feel like we could follow `toListL` implementation:
```
foldLeftL(mutable.ListBuffer.empty[A])(_ += _).map(_.toList)
```

I expect `map(_.toList)` to be much cheaper than appending to the end of list each time

File: monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala [hunk 2]
```diff
@@ -227,6 +227,17 @@ object Consumer {
   def raiseError[In, R](ex: Throwable): Consumer.Sync[In, R] =
     new RaiseErrorConsumer(ex)
 
+  /** A consumer that will return a collection of all the streamed
+   * values as a Scala `List`.
+   *
+   * Note that this method is functionally the same as the `.toListL` from [[Observable]]
```
Comment (Avasil) [inline monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala]: A link would be nice, i.e. `[[Observable.toListL]]`

File: monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala [hunk 3]
```diff
@@ -227,6 +227,17 @@ object Consumer {
...
+   *
+   * WARNING: For infinite streams the process will eventually
+   * blow up with an out of memory error.
+   */
+  def collect[A]: Consumer.Sync[A, List[A]] =
```
Comment (Avasil) [inline monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala]: `collect` usually uses partial function - what do you think about a name like `toList` (consistent with `toListL`) ?

Comment (paualarco) [reply monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala:238]: sure, btw, what does represent the final `L` from the `.toListL` ?

Comment (paualarco) [reply monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala]: amended all your comments, thank you for the quick feedback! :)

File: monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala [hunk 4]
```diff
@@ -227,6 +229,17 @@ object Consumer {
...
 
+  /** A consumer that will return a collection of all the streamed
+   * values as a Scala `List`.
+   *
+   * Note that this method is functionally the same as the [[Observable.toListL]]
+   *
+   * WARNING: For infinite streams the process will eventually
+   * blow up with an out of memory error.
+   */
+  def toListL[A]: Consumer[A, List[A]] =
```
Comment (Avasil) [inline monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala:240]: `L` is only used in `Observable` to differentiate between methods returning `Task` and `Observable`, I would leave just `toList` in `Consumer`. Sorry for nitpick :D

Comment (paualarco) [reply monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala]: No worries, I have changed that already :)

File: monix-reactive/shared/src/test/scala/monix/reactive/consumers/ListConsumerSuite.scala [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+import scala.util.{Failure, Success}
+
+object ListConsumerSuite extends TestSuite[TestScheduler] {
+  def setup(): TestScheduler = TestScheduler()
+  def tearDown(s: TestScheduler): Unit = {
+    assert(s.state.tasks.isEmpty, "TestScheduler should have no pending tasks")
+  }
+
+  test("should return the same all consumed elements as a sequence") { implicit s =>
```
Comment (Avasil) [inline monix-reactive/shared/src/test/scala/monix/reactive/consumers/ListConsumerSuite.scala:34]: I think it needs `s.tick()` for tests to pass and it's good to merge :)

Comment (paualarco) [reply monix-reactive/shared/src/test/scala/monix/reactive/consumers/ListConsumerSuite.scala]: thought it was passing already... 
is there a better way of running minitests than "sbt project reactiveTests" andThen "sbt test"?

Comment (Avasil) [reply monix-reactive/shared/src/test/scala/monix/reactive/consumers/ListConsumerSuite.scala]: `ReactiveTests` is something else but I can see how it is confusing :D 

```scala
reactiveJVM/test
```

or

```scala
reactiveJVM/testOnly *NameOfTheTestClass
```

**Pull request merged**